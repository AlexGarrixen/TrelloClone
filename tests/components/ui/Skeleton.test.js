import React from 'react';
import { mount } from 'enzyme';
import {
  Skeleton,
  SkeletonCol,
  SkeletonPicture,
  SkeletonRow,
} from '../../../src/components/ui/Skeleton';

describe('<Skeleton />', () => {
  test('Render sucess', () => {
    const wrapper = mount(<Skeleton />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton').exists()).toBe(true);
  });

  test('Test with prop spacing', () => {
    const wrapper = mount(<Skeleton spacing={3} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton').hasClass('space-3')).toBe(true);
  });
});

describe('<SkeletonPicture />', () => {
  test('Render sucess', () => {
    const wrapper = mount(<SkeletonPicture />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton__picture').exists()).toBe(true);
  });

  test('Test with prop cols', () => {
    const wrapper = mount(<SkeletonPicture cols={8} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton__picture').hasClass('col-8')).toBe(true);
  });
});

describe('<SkeletonRow />', () => {
  test('Render sucess', () => {
    const wrapper = mount(<SkeletonRow />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton__row').exists()).toBe(true);
  });
});

describe('<SkeletonCol />', () => {
  test('Render sucess', () => {
    const wrapper = mount(<SkeletonCol />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton__col').exists()).toBe(true);
  });

  test('Test with prop cols', () => {
    const wrapper = mount(<SkeletonCol cols={4} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.skeleton__col').hasClass('col-4')).toBe(true);
  });
});
