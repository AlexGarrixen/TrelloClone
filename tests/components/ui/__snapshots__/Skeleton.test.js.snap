// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> Render sucess 1`] = `
<div
  className="skeleton"
/>
`;

exports[`<Skeleton /> Test with prop spacing 1`] = `
<div
  className="skeleton space-3"
/>
`;

exports[`<SkeletonCol /> Render sucess 1`] = `
<div
  className="skeleton__col"
/>
`;

exports[`<SkeletonCol /> Test with prop cols 1`] = `
<div
  className="skeleton__col col-4"
/>
`;

exports[`<SkeletonPicture /> Render sucess 1`] = `
<div
  className="skeleton__picture"
/>
`;

exports[`<SkeletonPicture /> Test with prop cols 1`] = `
<div
  className="skeleton__picture col-8"
/>
`;

exports[`<SkeletonRow /> Render sucess 1`] = `
<div
  className="skeleton__row"
/>
`;
